cloud/clouds.go

```diff
@@ -156,6 +156,7 @@ type cloudSet struct {
 
 // cloud is equivalent to Cloud, for marshalling and unmarshalling.
 type cloud struct {
+	Name             string                 `yaml:"name"`
```

Comment:
Ah, actually that's a mistake - we should only be serialising the name outside of the clouds.yaml file format. i.e. clouds.yaml should continue to be formatted as:

```yaml
name:
    type: foo
...
```

and for other cases of YAML serialisation, we'll get:

```yaml
name: foo
type: bar
...
```

I have updated the code and added tests.